### PR TITLE
sage.{algebras,combinat,matroids}: Replace imports from sage.*.all for namespace packages

### LINKED ISSUE #34946: sage.{algebras,combinat,matroids}: Replace imports from sage.*.all for namespace packages
<div id="comment:0"></div>

Using `./sage -fiximports` from #34945.

Also remove trailing whitespace in the affected files.

Part of Meta-ticket #32414

CC:  @alexchandler100 @fchapoton @tscrim

Component: **refactoring**

Author: **Alex Chandler, Matthias Koeppe**

Reviewer: **Travis Scrimshaw**

_Issue created by migration from https://trac.sagemath.org/ticket/34946_

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description

Fixes #34946

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tscrim — CHANGES_REQUESTED** [review]
Two basically trivially things if you want to do them:

1. The imports `from sage.arith.misc import GCD as gcd` are somewhat unnecessary (like the `Q as QQ`).
2. The `FiniteField as GF` in the matroids, similarly. I am wondering if we should just replace the `GF` with `FiniteField`.

**Comment by mkoeppe** [comment]
> * The imports `from sage.arith.misc import GCD as gcd` are somewhat unnecessary (like the `Q as QQ`).

Thanks, indeed `gcd` is also provided by `sage.arith.misc`; so I've made this simplification.

> * The `FiniteField as GF` in the matroids, similarly. I am wondering if we should just replace the `GF` with `FiniteField`.

I'll not do this one here; importing it as GF seems like a convenient abbreviation

**Review by tscrim — APPROVED** [review]
Fair enough. LGTM.

**Comment by mkoeppe** [comment]
Thank you!

File: src/sage/combinat/species/cycle_species.py [hunk 1]
```diff
@@ -12,11 +12,13 @@
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
 
+from sage.arith.misc import divisors, euler_phi
+from sage.structure.unique_representation import UniqueRepresentation
+
+from .misc import accept_size
```
Comment (tobiasdiez) [inline src/sage/combinat/species/cycle_species.py]: Same comment about relative imports as in the other PR(s)

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

File: src/sage/combinat/species/cycle_species.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/combinat/species/cycle_species.py]: Done!

File: src/sage/matroids/linear_matroid.pyx [hunk 2]
```diff
@@ -129,7 +129,10 @@ from sage.matrix.matrix2 cimport Matrix
 import sage.matrix.constructor
 from sage.matrix.constructor import matrix
 from copy import copy, deepcopy
-from sage.rings.all import ZZ, QQ, FiniteField, GF
+from sage.rings.integer_ring import ZZ
+from sage.rings.rational_field import QQ
+from sage.rings.finite_rings.finite_field_constructor import FiniteField
+from sage.rings.finite_rings.finite_field_constructor import FiniteField as GF
```
Comment (tobiasdiez) [inline src/sage/matroids/linear_matroid.pyx]: These imports seem to be duplicates

Comment (mkoeppe) [reply src/sage/matroids/linear_matroid.pyx]: Yes, and the first one was not even used. I've cleaned it up now

**Review by tobiasdiez — APPROVED** [review]

**Pull request merged**